src/main/java/nl/thehyve/ocdu/controllers/TemplateController.java

```diff
@@ -52,16 +57,30 @@
             String url = user.getOcEnvironment();
             List<ClinicalData> clinicalDatas = clinicalDataRepository.findBySubmission(uploadSession);
 
-            //TODO: still not working, check the SOAP call
+            //key: subject id from user - val: technical subject id
             Map<String, String> subjectMap = openClinicaService.createMapSubjectLabelToSubjectOID(username, pwdHash, url, clinicalDatas);
-            List<String> test = new ArrayList<>();
-            for (String key : subjectMap.keySet()) {
-                test.add(key);
-            }
-            return new ResponseEntity<>(test, HttpStatus.OK);
+
+            Study study = dataService.findStudy(uploadSession.getStudy(), user, pwdHash);
+            MetaData metadata = openClinicaService.getMetadata(username, pwdHash, user.getOcEnvironment(), study);
+            PatientDataFactory pdf = new PatientDataFactory(user, uploadSession);
+            List<String> result = pdf.generatePatientRegistrationTemplate(metadata, subjectMap, registerSite);
+
+            return new ResponseEntity<>(result, HttpStatus.OK);
         } catch (Exception e) {
             e.printStackTrace();
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
         }
     }
+
+    @RequestMapping(value = "/upload-subjects", method = RequestMethod.POST)
+    public ResponseEntity<List<ValidationErrorMessage>> uploadSubjects(@RequestParam("newSubjects") MultipartFile newSubjectsFile, HttpSession session) {
+        try {
+//            newSubjectsFile.getInputStream().
```

Comment:
this was from my unfinished coding on the uploading of subjects, but will remove it